common/openshift_ops.py

```diff
@@ -0,0 +1,103 @@
+import logging
+from kubernetes import config
+from openshift.dynamic import DynamicClient
+
+log = logging.getLogger(__name__)
+
+
+class Ocp(object):
+    """Class which contains various utility functions for interacting
+    with OpenShift
+
+    """
+    def __init__(self):
+
+        k8s_client = config.new_client_from_config()
+        dyn_client = DynamicClient(k8s_client)
+
+        self.v1_service_list = dyn_client.resources.get(api_version='v1',
+                                                        kind='ServiceList')
+        self.v1_projects = dyn_client.resources.get(
+            api_version='project.openshift.io/v1', kind='Project')
+
+    def get_projects(self):
+        """Gets all the projects in the cluster.
+
+        Returns:
+            list: List of projects
+            NoneType: on failure
+        """
+        projects = []
+        try:
+            ret = self.v1_projects.get()
+            for each in ret.items:
+                projects.append(each.metadata.name)
+            return projects
+        except Exception as err:
```

Comment:
Any reason for such a broad exception?

Reply:
Catching all unknown exception at this point and logging the full exception to the log and returning None value, so that in any test case which involves this function return None if exception is caught, accordingly test case owner willl act ( either pass or fail ). Since full exception is logged, its easy to debug why its failed

Reply:
lets reraise the exception rather than return none, eventually it will fail but we can see how to better handle.

Reply:
Addressed in commit d859c97